libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Network/Network/Queue/CSFNetwork+Salesforce.m

```diff
@@ -39,17 +39,18 @@ - (void)setDefaultConnectCommunityId:(NSString *)defaultConnectCommunityId {
 - (void)setupSalesforceObserver {
     [[NSNotificationCenter defaultCenter] addObserver:self
                                              selector:@selector(userAccountManagerDidChangeCurrentUser:)
-                                                 name:SFUserAccountManagerDidChangeCurrentUserNotification
+                                                 name:SFUserAccountManagerDidChangeUserDataNotification
                                                object:nil];
 }
 
 #pragma mark SFAuthenticationManagerDelegate
-
 - (void)userAccountManagerDidChangeCurrentUser:(NSNotification*)notification {
     SFUserAccountManager *accountManager = (SFUserAccountManager*)notification.object;
-    if ([accountManager isKindOfClass:[SFUserAccountManager class]]) {
+    SFUserAccountChange change = (SFUserAccountChange)[notification.userInfo[SFUserAccountManagerUserChangeKey] integerValue];
+    if ([accountManager isKindOfClass:[SFUserAccountManager class]]
+            && (change & SFUserAccountChangeCurrentUser)) {
         if ([accountManager.currentUserIdentity isEqual:self.account.accountIdentity] &&
-            ![accountManager.currentCommunityId isEqualToString:self.defaultConnectCommunityId])
+                ![accountManager.currentCommunityId isEqualToString:self.defaultConnectCommunityId])
```

Comment:
As mentioned before, it would be better to be able to do a simple user-to-user comparison, by passing the user account in the notification.